src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -137,6 +187,131 @@ bool Terminal::EraseCharacters(const unsigned int numChars)
     return true;
 }
 
+// Method description:
+// - erases a line of text, either from
+// 1. beginning to the cursor's position
+// 2. cursor's position to end
+// 3. beginning to end
+// - depending on the erase type
+// Arguments:
+// - the erase type
+// Return value:
+// - true if succeeded, false otherwise
+bool Terminal::EraseInLine(const ::Microsoft::Console::VirtualTerminal::DispatchTypes::EraseType eraseType)
+{
+    const auto cursorPos = _buffer->GetCursor().GetPosition();
+    const auto viewport = _GetMutableViewport();
+    COORD startPos = { 0 };
+    startPos.Y = cursorPos.Y;
+    // nlength determines the number of spaces we need to write
+    DWORD nlength = 0;
+
+    // Determine startPos.X and nlength by the eraseType
+    switch (eraseType)
+    {
+    case DispatchTypes::EraseType::FromBeginning:
+        nlength = cursorPos.X - viewport.Left() + 1;
+        break;
+    case DispatchTypes::EraseType::ToEnd:
+        startPos.X = cursorPos.X;
+        nlength = viewport.RightInclusive() - startPos.X;
+        break;
+    case DispatchTypes::EraseType::All:
+        startPos.X = viewport.Left();
+        nlength = viewport.RightInclusive() - startPos.X;
+        break;
+    case DispatchTypes::EraseType::Scrollback:
+        return false;
+    }
+
+    auto eraseIter = OutputCellIterator(L' ', _buffer->GetCurrentAttributes(), nlength);

```

Comment:
I think L' ' should be UNICODE_SPACE defined by one of the headers you already have included.

Reply:
Hm it seems that UNICODE_SPACE is not defined in any of the headers though (I get an error saying its undefined), should there be some other file included in here?

Reply:
It's in `src/inc/unicode.hpp`

Reply:
Done!